### PR TITLE
Adding h3ronpy

### PR DESCRIPTION
**Pull request by keewis**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] ~Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md)~.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by keewis** [comment]
I'll assume that we can simply define [outputs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section) or create a new feedstock should we decide to publish `h3arrow` as a separate package and will try to go ahead with getting this PR in shape for reviews and possibly merging.

**Comment by kylebarron** [comment]
> I'll assume that we can simply define [outputs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section) or create a new feedstock should we decide to publish `h3arrow` as a separate package and will try to go ahead with getting this PR in shape for reviews and possibly merging.

I personally don't think we need to publish h3arrow to conda. It's a rust-only crate, right? Seems to be fine having that on crates.io?

**Comment by keewis** [comment]
> Seems to be fine having that on crates.io?

Most likely, yes.

I've removed @nmandery from the maintainers list for the sake of asking for reviews and merging, but I'm very open to reverting that, either here or on the feedstock once that has been created.

**Comment by keewis** [comment]
@conda-forge/help-rust, ready for review!

**Comment by xhochy** [comment]
Please only ask for review once the CI has passed.

**Comment by keewis** [comment]
it did pass before I removed a maintainer (the CI flag is red on that commit because there was no confirmation from them), so I'm pretty confident it will pass again

File: recipes/h3ronpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - h3ronpy.pandas.raster
+    - h3ronpy.pandas.vector
+    - h3ronpy.polars
+  commands:
+    - pip check
+  requires:
+    - pip
+    - geopandas
+    - pyarrow
+    - polars
```
Comment (xhochy) [inline recipes/h3ronpy/meta.yaml:52]: Either make then `run` dependencies or omit the optional modules from the import check
```suggestion
```

File: recipes/h3ronpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - pyarrow >=15
+    - polars >=1
+
+test:
+  imports:
+    - h3ronpy
+    - h3ronpy.pandas
+    - h3ronpy.pandas.raster
+    - h3ronpy.pandas.vector
+    - h3ronpy.polars
```
Comment (keewis) [inline recipes/h3ronpy/meta.yaml:45]:
```suggestion
```

**Review by xhochy — APPROVED** [review]

**Comment by keewis** [comment]
Thanks for the reviews, @xhochy!

**Pull request merged**